install_config/configure_kerberos_for_sso.adoc

```diff
@@ -0,0 +1,174 @@
+==== Configuring an Apache proxy or Single Sign On authentication via Kerberos
+
+Configuring SSO authentication makes life much easier for users that are
+enrolled in a Windows Active Directory domain or a Linux IPA domain. Both these
+enteprise authentication systems use Kerberos and that allows a seamless
+experience where the user doesn't need to use additional passwords to
+authenticate to systems and services.
+
+The key challenge in getting this setup to work properly is correct
+configuration of domains names. Kerberos realies on correct DNS resolution and
+fully qualified domain names to resolve how to acquire the proper token to
+authenticate to a network service. Naming errors are the leading cause of
+unsuccessful deployments.
+
+===== Installation prerequisites
+
+First establish the (public) DNS name that you are going to assign to the
+Apache Authentication Proxy. This DNS name will be used in your domain
+controller (AD, IPA, or other Kerberos Distribtuon Center) to create a new
+Kerberos Principal and Keys associated with the proxy. These keys will be
+then stored into a keytab that will be used by Apache to authenticate users.
+No other name or alias will work, so make sure to always reference and use the
+fully qualified DNS name you choose for this proxy and all relevant cluster
+configurations.
+
+Obtain a keytab for the HTTP Service Principal Name (SPN) associated to the
+chosen DNS name from your KDC (AD, IPA, other). The SPN to be created is a
+two component principal where the first component is the string `HTTP` (all
```

Comment:
Can I have a bit more context about "two component principal"?

II think I'd reorder this to be:
1) Make the SPN
2) Assign a key
3) Obtain a keytab for the SPN.

Are these three actions standard practices that admins are familiar with, or do we need to provide more information about completing them?

Reply:
SPN is defined above, and it is a standard acronym admins that know a minimum of kerberos should be familiar with.
A two compoenent principal is any primcipal that gets stringified as comp1 / comp2 @ REALM I may avoid mentioning "two componetns" though, it doesn't help clarity anyway.